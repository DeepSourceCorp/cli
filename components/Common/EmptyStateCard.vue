<template>
  <section
    class="empty-state-card-shadow mx-auto flex w-full flex-wrap justify-center gap-x-4 gap-y-1 rounded-md border border-solid border-ink-200 bg-ink-300 px-0 py-6 md:flex-nowrap"
  >
    <empty-state-picture
      :webp-image-path="webpImagePath"
      :png-image-path="pngImagePath"
      :svg-image-path="svgImagePath"
      :width="imageWidth ? imageWidth : 'w-auto'"
      :alt-text="altText"
      :height="imageWidth ? false : 'h-17 md:h-20'"
      :image-classes="imageClasses"
      class="w-full md:w-auto"
    />

    <div :class="contentStyles">
      <h3 class="mt-4 text-base font-medium text-vanilla-100 md:mt-0">
        <slot name="title">
          {{ title }}
        </slot>
      </h3>

      <p class="mt-2 text-sm leading-6 text-vanilla-400">
        <slot name="subtitle">
          {{ subtitle }}
        </slot>
      </p>
    </div>
  </section>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'nuxt-property-decorator'

@Component({ name: 'EmptyStateCard' })
export default class EmptyStateCard extends Vue {
  @Prop({ default: 'This page has no data' })
  altText: string

  @Prop({ default: 'text-left px-4 max-w-sm', type: String })
  contentStyles: string

  @Prop({ default: 'md:mx-auto px-4 md:px-0', type: String })
  imageClasses: string

  @Prop()
  imageWidth: string

  @Prop({ default: '' })
  subtitle: string

  @Prop({ default: '' })
  title: string

  @Prop({ default: require('~/assets/images/ui-states/empty.gif') })
  pngImagePath: () => any

  @Prop({ default: '' })
  svgImagePath: () => any

  @Prop({ default: require('~/assets/images/ui-states/empty.webp') })
  webpImagePath: () => any
}
</script>

<style scoped>
.empty-state-card-shadow {
  box-shadow: 0px 4px 16px rgba(0, 0, 0, 0.25);
}
</style>
