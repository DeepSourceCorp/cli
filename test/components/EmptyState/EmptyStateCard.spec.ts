import { render } from '@testing-library/vue'

import EmptyStateCard from '~/components/Common/EmptyStateCard.vue'

describe('[[ EmptyStateCard ]]', () => {
  const propsData = {
    subtitle: 'Test subtitle',
    title: 'Test title'
  }

  const stubs = {
    EmptyStatePicture: true
  }

  test('renders the empty state card pattern', () => {
    const { html } = render(EmptyStateCard, { propsData, stubs })
    expect(html()).toMatchSnapshot()
  })

  test('renders empty state card pattern with content supplied via slots', () => {
    const { html } = render(EmptyStateCard, {
      slots: propsData,
      stubs
    })

    expect(html()).toMatchSnapshot()
  })
})
