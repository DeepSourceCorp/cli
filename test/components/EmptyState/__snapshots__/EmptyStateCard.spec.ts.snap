// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[[ EmptyStateCard ]] renders empty state card pattern with content supplied via slots 1`] = `
"<section class=\\"empty-state-card-shadow mx-auto flex w-full flex-wrap justify-center gap-x-4 gap-y-1 rounded-md border border-solid border-ink-200 bg-ink-300 px-0 py-6 md:flex-nowrap\\">
  <emptystatepicture-stub webp-image-path=\\"test-image-dir/test-image-file.png\\" png-image-path=\\"test-image-dir/test-image-file.png\\" svg-image-path=\\"\\" width=\\"w-auto\\" alt-text=\\"This page has no data\\" height=\\"h-17 md:h-20\\" image-classes=\\"md:mx-auto px-4 md:px-0 w-full md:w-auto\\"></emptystatepicture-stub>
  <div class=\\"text-left px-4 max-w-sm\\">
    <h3 class=\\"mt-4 text-base font-medium text-vanilla-100 md:mt-0\\">Test title</h3>
    <p class=\\"mt-2 text-sm leading-6 text-vanilla-400\\">Test subtitle</p>
    <div class=\\"mt-6\\">Action on!</div>
  </div>
</section>"
`;

exports[`[[ EmptyStateCard ]] renders the empty state card pattern 1`] = `
"<section class=\\"empty-state-card-shadow mx-auto flex w-full flex-wrap justify-center gap-x-4 gap-y-1 rounded-md border border-solid border-ink-200 bg-ink-300 px-0 py-6 md:flex-nowrap\\">
  <emptystatepicture-stub webp-image-path=\\"test-image-dir/test-image-file.png\\" png-image-path=\\"test-image-dir/test-image-file.png\\" svg-image-path=\\"\\" width=\\"w-auto\\" alt-text=\\"This page has no data\\" height=\\"h-17 md:h-20\\" image-classes=\\"md:mx-auto px-4 md:px-0 w-full md:w-auto\\"></emptystatepicture-stub>
  <div class=\\"text-left px-4 max-w-sm\\">
    <h3 class=\\"mt-4 text-base font-medium text-vanilla-100 md:mt-0\\">
      Test title
    </h3>
    <p class=\\"mt-2 text-sm leading-6 text-vanilla-400\\">
      Test subtitle
    </p>
    <!---->
  </div>
</section>"
`;
